### DISCUSSION TITLE
What is recommended way to create and analyze memory dump from ASP.NET Core 8 application running in docker container?

### DISCUSSION THREAD

**Discussion by rouen-sk** [dotnet/dotnet-docker#5232] [Q&A] [answered]
I have an ASP.NET Core 8 application, containerized (using MSBuild, without dockerfile) running in AKS. It is stable (not crashing, not leaking) but the memory usage is higher than expected. So I wanted to investigate, using dotnet-counters and dotnet-dump.

I've managed to connect to pod shell
```
kubectl exec -n prod --stdin --tty my-api-77cff99b66 -- /bin/bash
```
but since the base image is ASP.NET Core runtime only, tools like `dotnet-counters` and `dotnet-dump` are not present, and cannot be installed. User is not root, there is no sudo, therefore no apt install.

What is recommended way to create and analyze dump from production application running on runtime (not SDK) image? Is it even possible? Or do I need to rebuild image with SDK base image and/or root user, and deploy it to production? 

Thank you!

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by lbussell** [comment, marked as answer]
[Triage] We recommend running .NET Monitor in your Kubernetes cluster in order to collect the dump from your application. There's some documentation on how to set up .NET Monitor with Kubernetes [here](https://github.com/dotnet/dotnet-monitor/blob/main/documentation/kubernetes.md). .NET Monitor images are published out of this repo: https://github.com/dotnet/dotnet-docker/blob/main/README.monitor.md.
